### PR TITLE
Update load api

### PR DESCRIPTION
**Pull request by jja725**
What changes are proposed in this pull request?
couple minor updates to load api

Why are the changes needed?
code quality

Does this PR introduce any user facing changes?
na

### CODE REVIEW

File: core/common/src/main/java/alluxio/retry/RetryUtils.java [hunk 1]
```diff
@@ -70,6 +70,9 @@ public static <V> V retryCallable(String description, Callable<V> f, RetryPolicy
       } catch (Exception e) {
         LOG.warn("Failed to {} (attempt {}): {}", description, policy.getAttemptCount(),
             e.toString());
+        if (e instanceof AlluxioRuntimeException && !((AlluxioRuntimeException) e).isRetryable()) {
```
Comment (jja725) [inline core/common/src/main/java/alluxio/retry/RetryUtils.java:73]: would not retry if we know it's not retryable

File: core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java [hunk 2]
```diff
@@ -220,6 +220,19 @@ public void abortBlock() throws Exception {
     assertFalse(mBlockWorker.getBlockStore().getTempBlockMeta(blockId).isPresent());
   }
 
+  @Test
+  public void createBlockAfterAbort() throws Exception {
```
Comment (jja725) [inline core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java:224]: additional test case

File: core/server/worker/src/main/java/alluxio/worker/block/UfsIOManager.java [hunk 3]
```diff
@@ -205,7 +209,7 @@ private int readInternal() {
                 .setPositionShort(mOptions.getPositionShort()));
         while (bytesRead < mLength) {
           int read;
-          read = inStream.read(mBuffuer.array(), bytesRead, (int) (mLength - bytesRead));
+          read = Channels.newChannel(inStream).read(mBuffuer);
```
Comment (jja725) [inline core/server/worker/src/main/java/alluxio/worker/block/UfsIOManager.java:212]: use channel to better aligh with the bytebuffer usage

File: core/server/common/src/main/java/alluxio/RpcUtils.java [hunk 4]
```diff
@@ -174,8 +174,9 @@ public static <T> void invoke(Logger logger, CompletableFuture<T> future, String
           responseObserver.onCompleted();
         } else {
           MetricsSystem.counter(getQualifiedFailureMetricName(methodName)).inc();
+          logger.warn("Exception when invoking : {}: {}", methodName, debugDesc, t);
           responseObserver.onError(
-              AlluxioRuntimeException.from(t.getCause()).toGrpcStatusRuntimeException());
+              AlluxioRuntimeException.from(t).toGrpcStatusRuntimeException());
```
Comment (jja725) [inline core/server/common/src/main/java/alluxio/RpcUtils.java:179]: whenComplete throw the original exception so getCause is wrong.

**Review by beinan — APPROVED** [review]
lgtm, thanks!

**Comment by beinan** [comment]
alluxio-bot, merge this please.

**Comment by jiacheliu3** [comment]
@jja725  @Xenorith Just a sanity check, any doc update needed for the new load api?

**Comment by jja725** [comment]
@jiacheliu3 no need for doc right now

**Pull request merged**